ReSwift/CoreTypes/ActionCreator.swift

```diff
@@ -10,9 +10,18 @@ import Foundation
 
 public protocol ActionCreator {
     
-    associatedtype State
+    associatedtype State: StateType
+    /**
+     Generates one or more new actions from the current state.  May be implemented asynchronously or synchronously
+     - parameter actionBlock This may be called more than once with different actions
+     */
+    func createActions(currentState state: State, actionBlock:(Action -> Void))
```

Comment:
Does the ActionCreator not need a reference to a dispatcher? This is OK if the store is a global var but we may want to move away from that